proposals/service-binding/service-binding.adoc

```diff
@@ -0,0 +1,101 @@
+
+=== Workflow
+
+Two main workflows have been identified in the specification that allows user applications to bind to a _Provisioned Service_. Let's review them and see which one would suit Camel-k the best.
+
+The first way outlined in the specification is for the user to create a `ServiceBinding` indicating which _application_ to bind to which set of _Provisioned Services_:
+
+[source,yaml]
+----
+kind: ServiceBinding
+metadata:
+  name: account-service
+spec:
+  application:
+    apiVersion: apps/v1
+    kind:       Deployment
+    name:       camel-k-integration
+
+  service:
+    apiVersion: com.example/v1alpha1
+    kind:       Database
+    name:       my-database
+----
+
+Once created, it is the responsibility of the _Service Binding Operator_ to inject the binding information into the _application_. The _application_  is any `PodSpec`-able resource e.g a `Deployment`. The way it does this is that it first scales the `Deployment` down to 0. Then it injects the binding information into the `Deployment` in the form of a `Secret` or a `ConfigMap`. Finally it scales the `Deployment` back up.
+
+The second way is through https://github.com/k8s-service-bindings/spec#custom-projection-service-binding-example-resource[Custom Projection]. The difference between this approach and the first one is that the binding information injection into the _application_ is not the responsibility of the _Service Binding Operator_ anymore. It wil be the responsibility of another Operator. In our case it will be Camel-k's responsibility to retrieve the binding information and inject it into the `Integration`.
+
+The drawback of the first approach, in the context of Camel-k, is that the ``Integration``'s `Deployment` would be created in an incomplete state and would result in an error the first time the underlying container is started because the binding information is not present. The second drawback is that the ``Integration``'s `Deployment` resource should ideally _not_ be altered by someone other than the Camel-k operator itself. The advantage of using this approach is that less work is needed in Camel-k as injection is taken care of by the _Service Binding Operator_.
+
+Conversely, the drawback of implementing the second approach is that more work is needed as Camel-k would have to retrieve and inject the binding information.
+The advantage would be that Camel-k retains full control of the ``Integration``'s `Deployment/Cron/Knative` resource and we can delay the startup until the binding information is available.
+
+For these reasons, it seems that the second approach is preferred in the context of Camel-k.
```

Comment:
assuming the camel-k operator is installed at global scope, using this approach would require the camel-k operator to have basically have access to any resource in any namespaces ?

Reply:
Thanks for reviewing! It would require the camel-k operator to have `read` access to the `ServiceBinding` and it's corresponding `Secret`. It would also need to have rights to `create` a `ServiceBinding`

Reply:
does the camel-k operator really need to access the secret ? I mean, the operator should add the service binding secret among the list of the secrets the integration needs, nothing more right ?

Reply:
no it doesn't, I added it for convenience to create property mappings like `{DB_PORT}` instead of `{secret:name/DB_PORT}`. Doesn't camel-k already have rights to read `Secret`s ? 
Nevertheless I think we can achieve this by adding the logic in camel-k-runtime via a customizer and avoid camel-k reading the `Secret`.
Ideally though, I think we would want to add some prefix to the property to avoid clashes like `{binding:name/DB_PORT}` where the user could elegantly choose the name

Reply:
yes, you can tell camel-k to mount a secret and make its content available as properties

Reply:
> Ideally though, I think we would want to add some prefix to the property to avoid clashes like `{binding:name/DB_PORT}` where the user could elegantly choose the name

I'm not very sure about this one as the main goal is to be able to auto configure components without the user intervention, on quarkus one of the option that has been taken into account is to have a binding aware config source that is in charge to translate binding parameters to the related quarkus property and I think we should do something similar for components.

If users want to directly reference binding parameters, they should use the same syntax as for secrets `{secret:$name/$property}` as it is already namespaced.

Reply:
> the main goal is to be able to auto configure components without the user intervention

The Camel Dream :+1: 

> on quarkus one of the option that has been taken into account is to have a binding aware config source that is in charge to translate binding parameters to the related quarkus property

This sounds promising, do you have a Camel or a Quarkus example I could look at ? I wasn't aware this was possible

Reply:
The reference issue is here: https://github.com/quarkusio/quarkus/issues/13617

As side note, for some components we may benefit directly from the work done in Quarkus as example, if the jdbc set-up is auto-magically done by a Quarkus extension, then we can consume it as we do as today without having anything to change (same could happen for kafka, infinispan, jms, etc.)

Reply:
That's definitely something we want to leverage :+1: I'll start hacking on this